cmd/object-handlers.go

```diff
@@ -183,40 +210,18 @@ func (api objectAPIHandlers) SelectObjectContentHandler(w http.ResponseWriter, r
 		return
 	}
 
-	getObject := objectAPI.GetObject
-	if api.CacheAPI() != nil && !crypto.SSEC.IsRequested(r.Header) {
-		getObject = api.CacheAPI().GetObject
-	}
-	reader, pipewriter := io.Pipe()
-
-	// Get the object.
-	var startOffset int64
-	length := objInfo.Size
-
-	var writer io.Writer
-	writer = pipewriter
+	// Set encryption response headers
 	if objectAPI.IsEncryptionSupported() {
```

Comment:
two if conditions for the same we don't need to do this we only need one..

Reply:
Fixing.